qoi.h

```diff
@@ -538,11 +551,17 @@ void *qoi_decode(const void *data, int size, qoi_desc *desc, int channels) {
 		else if (p < chunks_len) {
 			int b1 = bytes[p++];
 
-			if ((b1 & QOI_MASK_2) == QOI_INDEX) {
+			if ((b1 & QOI_MASK_4) == QOI_INDEX) {
 				px = index[b1 ^ QOI_INDEX];
 			}
-			else if ((b1 & QOI_MASK_2) == QOI_RUN) {
-				run = (b1 & 0x3f);
+			else if ((b1 & QOI_MASK_4) == QOI_RUN) {
+				run = (b1 & 0xf);
+				while (p < chunks_len && ((b1 = bytes[p]) & QOI_MASK_2) == QOI_RUN)
```

Comment:
QOI_MASK_4, right?

It might help to add a round-trip test like the qoibench.c change in
https://github.com/phoboslab/qoi/pull/47/files#diff-242b92f8574841292eb78ac67a94b42000dbfeb16a567e8e5e0994bb928df2f5

Reply:
that's really helpful! It's still not correct though. Time to do some debugging...